rslib/src/scheduler/answering/mod.rs

```diff
@@ -269,12 +269,17 @@ impl Collection {
 
         let mut updater = self.card_state_updater(card)?;
         answer.cap_answer_secs(updater.config.inner.cap_answer_time_to_secs);
-        let current_state = updater.current_card_state();
-        require!(
-            current_state == answer.current_state,
-            "card was modified: {current_state:#?} {:#?}",
-            answer.current_state,
-        );
+        let mut current_state = updater.current_card_state();
+        // If the states aren't equal, it's probably because some time has passed.
+        // Try to fix this by setting elapsed_time equal.
+        if answer.current_state != current_state {
+            self.set_elapsed_time_equal(answer, &mut current_state);
```

Comment:
We probably don't need the outer check, since set_elapsed_time_equal() will only apply to (re)learning cards.

Reply:
I figured it might save a few instructions depending on how the compiler figures it out, but yeah it's probably unnecessarcy.